string topic_name
string topic_type
string message
